pkg/agent/cniserver/interface_configuration_windows.go

```diff
@@ -181,26 +205,84 @@ func (ic *ifConfigurator) createContainerLink(endpointName string, result *curre
 	return hnsEP, nil
 }
 
+func (ic *ifConfigurator) getInterfacesConfigByPods(pods sets.String) map[string]*interfacestore.InterfaceConfig {
+	interfaces := make(map[string]*interfacestore.InterfaceConfig)
+	ic.epCache.Range(func(key, value interface{}) bool {
+		ep, _ := value.(*hcsshim.HNSEndpoint)
+		ifConfig := parseOVSPortInterfaceConfigFromHNSEndpoint(ep)
+		if ifConfig == nil {
+			return true
+		}
+		namespacedName := k8s.NamespacedName(ifConfig.PodNamespace, ifConfig.PodName)
+		if pods.Has(namespacedName) {
+			interfaces[namespacedName] = ifConfig
+		}
+		return true
+	})
+	return interfaces
+}
+
+func parseOVSPortInterfaceConfigFromHNSEndpoint(ep *hcsshim.HNSEndpoint) *interfacestore.InterfaceConfig {
+	portData := &ovsconfig.OVSPortData{
+		Name:        ep.Name,
+		ExternalIDs: ep.AdditionalParams,
+	}
+	ifaceConfig := ParseOVSPortInterfaceConfig(portData, nil)
+	if ifaceConfig != nil {
+		var err error
+		ifaceConfig.MAC, err = net.ParseMAC(ep.MacAddress)
+		if err != nil {
+			klog.Errorf("Failed to parse MAC address from OVS external config %s: %v", ep.MacAddress, err)
+			return nil
+		}
+	}
+	return ifaceConfig
+}
+
 // attachContainerLink takes the result of the IPAM plugin, and adds the appropriate IP
 // addresses and routes to the interface. It then sends a gratuitous ARP to the network.
 func attachContainerLink(ep *hcsshim.HNSEndpoint, containerID, sandbox, containerIFDev string) (*current.Interface, error) {
 	var attached bool
-	attached, err := ep.IsAttached(containerID)
-	if err != nil {
-		return nil, err
+	var err error
+	var hcnEp *hcn.HostComputeEndpoint
+	if sandbox == "none" || strings.Contains(sandbox, ":") {
```

Comment:
it would be good to have a comment to explain what these cases mean

Reply:
Done.

Reply:
Yes, it's different with `isInfraContainer `. I add a new function `isDockerContainer` for it and add more comments.